apps/server/src/modules/board/uc/element.uc.ts

```diff
@@ -40,10 +45,21 @@ export class ElementUc extends BaseUc {
 		return element;
 	}
 
-	async deleteElement(userId: EntityId, elementId: EntityId): Promise<void> {
+	async deleteElement(userId: EntityId, elementId: EntityId, auth = ''): Promise<void> {
 		const element = await this.getElementWithWritePermission(userId, elementId);
 
 		await this.elementService.delete(element);
+
+		if (element instanceof DrawingElement) {
+			await firstValueFrom(
+				this.httpService.delete(`${Configuration.get('TLDRAW_URI') as string}/tldraw-document/${element.drawingName}`, {
```

Comment:
dont directly do http calls from the uc. This should be done in a similar fashion to the file deletion. Please define an adapterService that exposes a deletion function for tldraw documents, and call it from the recursiveSaveVisitor. As this is only needed in the board, the adapter can be defined in the board/repo folder.

Im not entirely sure how to best solve authorisation in this case. Since the server has already authorized the operation, the deletion should always go through no matter what. Therefore, I think the call to tldraw should not be authorized with the users credentials, but rather with some administrative privileges granted to the server.

Finally, please use axios instaed of http directly

Reply:
I created adapter and moved there the logic. Deleted for now authorisation, but i have consern that probably we shouldn't leave this /tldraw-document/${element.drawingName} endpoint open without auth processing.